src/gui/dialog/SettingsDialog.cpp

```diff
@@ -188,6 +193,13 @@ void SettingsDialog::customHandRecognitionToggled() {
     gtk_widget_set_sensitive(get("boxCustomTouchDisableSettings"), touchMethod == 2);
 }
 
+void SettingsDialog::customStylusIconTypeChanged() {
+    GtkWidget* cbStylusCursorType = get("cbStylusCursorType");
+    int stylusCursorType = gtk_combo_box_get_active(GTK_COMBO_BOX(cbStylusCursorType));
+    bool showCursorHighlightOptions = stylusCursorType != STYLUS_CURSOR_NONE;
+    gtk_widget_set_visible(get("highlightCursorGrid"), showCursorHighlightOptions);
```

Comment:
Maybe it would be better to only make these settings insensitive than invisible.
```suggestion
    gtk_widget_set_sensitive(get("highlightCursorGrid"), showCursorHighlightOptions);
```